### PR TITLE
CRM-21456 : Relationship end date not set when creating resolved cases

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Set relationship end date while creating resolved cases.

Before
----------------------------------------
1. Relationship end date is not set when creating a new case with resolved status. 

2. When a Resolved case is created via api(through explorer or via webform). Case end date is not automatically stored. It is correctly set from the "New Case" form.

See more details in jira issue.

After
----------------------------------------
Both the above issues are fixed.

Technical Details
----------------------------------------
This PR migrates the case end date calculation from `Form` to `BAO` layer, so that it is executed from API too.

Comments
----------------------------------------
Added unit test

---

 * [CRM-21456: Relationship end date not set when creating resolved cases.](https://issues.civicrm.org/jira/browse/CRM-21456)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
jenkins, retest this please

File: CRM/Case/BAO/Case.php [hunk 1]
```diff
@@ -92,6 +92,11 @@ public static function &create(&$params) {
     // CRM-20958 - These fields are managed by MySQL triggers. Watch out for clients resaving stale timestamps.
     unset($params['created_date']);
     unset($params['modified_date']);
+    $caseStatus = CRM_Case_PseudoConstant::caseStatus('name');
```
Comment (omarabuhussein) [inline CRM/Case/BAO/Case.php]: please add a space before this line

**Comment by omarabuhussein** [comment]
Checked the code and did some UI testing and it looks okay, just left minor comment@jitendrapurohit

**Comment by eileenmcnaughton** [comment]
I'm going to merge this based on @omarabuhussein review. I realise there is a stylistic comment outstanding but I don't want to hold up merge on it as I'm looking at it now and I think it's more a suggestion than a requirement

**Pull request merged**